### DISCUSSION TITLE
[Phont Auth] not receive SMS

### DISCUSSION THREAD

**Discussion by fokoz** [invertase/react-native-firebase#6356] [General]
Real devices don't receive SMS without any errors. This problem just has been happening around 29 JUN 2022. Before this It worked fine  and I haven't changed any codes. What should I do ?
I am using `@react-native-firebase/auth v9.3.5`

**Comment by mikehardy** [comment]
There is no support provided for versions that old, sorry

Please update to current versions and see if it is still happening (here's a guide explaining our release process so you can adapt more easily to the changes: https://invertase.io/blog/react-native-firebase-versioning)

Note that you may be having an issue related to https://github.com/firebase/firebase-android-sdk/issues/2688 where perhaps the app name being too long causes an issue. That's unresolved but has interesting reading

First step is to get up to date though, as running software multiple years old on mobile devices should come with no expectation anything will work, the device operating systems move too quickly for that.

**Reply by fokoz** [reply]
It just doesn't work since 29 JUN, I just tested around 26 JUN and it worked fine. Is app name too long really  an issue here ?

Thank you

**Reply by mikehardy** [reply]
> There is no support provided for versions that old, sorry

**Reply by fokoz** [reply]
OK thank you for your quick response, I will try upgrading to the latest version.

**Comment by fokoz** [comment]
I have upgraded to v14.11.0 and the problem still occur. What should I do next ? Should I contact google team or may be the cellular provider in my country block firebase sms ?

**Reply by mikehardy** [reply]
Oh I am sorry to hear that, I was hoping having the updated SDKs would do the trick. I did notice in my own work projects that my SMS providers did sometimes block the SMS - we have three carriers in our country, 1 the SMS never came through for anyone, 1 the SMS came through but only if the phone number had never been ported (strange one! it was inconsistent between devices...), and 1 they came through most of the time but not all of the time. it was really frustrating, but it definitely felt to us like a carrier issue.

So try it with different people you know, if you can, that have different carriers and maybe have never ported their number. If that does not work, definitely watch adb logcat to see if any SMS comes through, and beyond that maybe approach firebase support yes.

Note that you are still not on the most up to date version if you are on v14.x. I understand getting iOS to work well with v15 is difficult, but you *can* actually use the most up to date firebase-android-sdk (v30.2.0 there) with an override and it works with 14.11.1 here - https://rnfirebase.io/#android - remember to take the override out later, but it's a way to get firebase-android-sdk support since you will have the most up to date code (they will likely ask for it...)

**Reply by fokoz** [reply]
I have a problem with v15 because of pod install failure according to this https://github.com/invertase/react-native-firebase/issues/6332

**Reply by mikehardy** [reply]
Yes I completely understand. I believe v15 can be made to work in nearly all situations now (note the bug title says "has workaround"...), *but it is not easy* and you already have one problem (on android), no need to go looking for more trouble on iOS.

Actually, that's an assumption - is your problem on both platforms, or just one? If it is on both perhaps it is some sort of throttling or payment issue. If it is just on one platform then I'm less sure - I was assuming android, but now that I re-read your comments I see you never specified, so my assumption has no real basis

**Reply by fokoz** [reply]
It seems that both OS have the problem. Just a second ago my team said that they received SMS again. What is going on ?

**Reply by fokoz** [reply]
But my phone still not receiving the SMS but I use different SMS provider from my team

**Reply by fokoz** [reply]
Is there any sms sending logs on firebase console ? Haven't seen any of them on the menu.

**Reply by mikehardy** [reply]
Probably the carrier. It is frustrating, yes. In Ecuador (where most of my users are) it is not reliable at all basically. Apparently the services that do have reliable delivery (the banks here) have worked out deals with the carriers. So their stuff works. Very few other people's stuff works reliably. Not firebase's fault as far as I can tell, it's a carrier thing

**Reply by fokoz** [reply]
Thank you so much for your very quick response again. 

I would have to try another third-party OTP to solve this thing.

Keep doing your great work !
